core/src/main/java/de/danoeh/antennapod/core/service/download/NewEpisodesNotification.java

```diff
@@ -0,0 +1,84 @@
+package de.danoeh.antennapod.core.service.download;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+import de.danoeh.antennapod.core.util.gui.NotificationUtils;
+
+public class NewEpisodesNotification {
+    static final String GROUP_KEY = "de.danoeh.antennapod.EPISODES";
+
+    private final int lastEpisodeCount;
+    private final boolean dontShowNotification;
+
+    public NewEpisodesNotification(Long feedId) {
+        Feed feed = DBReader.getFeed(feedId);
+
+        FeedPreferences prefs = feed.getPreferences();
+        if (!prefs.getKeepUpdated() || !prefs.getShowEpisodeNotification()) {
+            dontShowNotification = true;
+            lastEpisodeCount = -1;
+            return;
+        }
+
+        lastEpisodeCount = PodDBAdapter.getInstance().getFeedCounters(feedId).get(feedId);
+
+        dontShowNotification = false;
+    }
+
+    public void showIfNeeded(Context context, Feed feed) {
+        if (dontShowNotification) {
+            return;
+        }
+
+        long feedId = feed.getId();
+        int episodeCount = PodDBAdapter.getInstance().getFeedCounters(feedId).get(feedId);
```

Comment:
Even if the adapter's open and close methods currently don't do much, please use them for consistency.

```suggestion
        PodDBAdapter adapter = PodDBAdapter.getInstance();
        adapter.open();
        int episodeCount = adapter.getFeedCounters(feedId).get(feedId);
        adapter.close();
```